internal/driver/webhtml.go

```diff
@@ -736,35 +758,37 @@ function viewer(baseUrl, nodes) {
     elem.addEventListener('touchstart', updater);
 
     function updater() {
-      elem.href = updateUrl(new URL(elem.href), param);
+      // The selection can be in one of two modes: regexp-based or
+      // list-based.  Construct regular expression depending on mode.
+      let re = regexpActive
```

Comment:
Since you are changing the code around handling the filters, could you describe / confirm which manual testing is done? There are no automated tests for the web UI feature unfortunately.

Reply:
Yeah, I played with this for a little while and everything seemed correct (after my most recent cleanup). Here's an incomplete list of things I tested:
- All "Refine" options still work with empty and non-empty regexp without sample index
- All "Refine" options still work with empty and non-empty regexp with sample index
- "Reset" button works
- Sample index selection does not remove "Refine" options
- "Refine" option does not remove sample index selection
- No errors in the web console
- Sample button not show when profile only has one sample